editor.planx.uk/src/@planx/graph/index.ts

```diff
@@ -158,6 +161,13 @@ const _add = (
 
   draft[id] = sanitize(nodeData);
 
+  if (isSectionNodeType(id, draft) && parent !== ROOT_NODE_KEY) {
+    alert(
+      "cannot add sections on branches or in portals, must be on center of main graph. close this window & try again"
+    );
+    throw new Error("cannot add sections on branches or in portals");
+  }
```

Comment:
when calling `move()` or `clone()`, thrown errors are immediately raised as user-facing alert dialogs - but `_add()` only logs them, so I've additionally added an alert here. 

it'd be most ideal to automatically close the section dialog in the Editor on error, but i think this should be fine for now!

Reply:
Going to note this one in a future card, and merge as-is - but appreciate these ideas!

My quick attempt at Introducing `navigate()` into graph/index methods is throwing invalid hooks errors and feels a little wrong still (like _should_ navigation/routing be a responsibility of these methods?). And I don't feel quite ready to figure out how to render the error in the Editor modal yet!